### ISSUE TITLE
Returning from HDR at end of file fails

### ISSUE THREAD

**Issue by roidy** [jellyfin-archive/jellyfin-roku-legacy#1567]
## Software Versions

- Jellyfin Server Version:  10.8.10
- Roku Client Version:  2.0 build 0

## Describe the bug
When playing any HDR file, either HDR10 or Dolby vision and pressing the back button on the remote control the video stops and the Roku returns to SDR mode. However if you let the file play to the end and the video naturally stops on it's own the Roku remains in HDR mode when returning to the interface.

This is using a Streaming Stick 4k model 3820EU and has been tested on two different TV's. Also the Roku's own media player handles this fine, the Roku returns to SDR mode regardless of how you stop the file, from the 'back' button or letting it play to the end.

## How To Reproduce

1. Play any HDR video
2. Press 'back' to stop the video and the Roku returns to SDR mode
3. Play any HDR video
4. Allow the video to stop naturally at the end and the Roku remains in HDR mode and doesn't return to SDR mode

## Expected behavior

Regardless of weather you manually stop a video of allow it to stop at the end the Roku should return to SDR mode.

## Connection Information

- Is server local or remote? Local
- Is server connection HTTP or HTTPS? HTTP

**Comment by cewert** [comment]
Is this no longer an issue for you?

**Comment by roidy** [comment]
Sorry didn't mean to close this issue.

Still a problem but after some investigating it seems the player state "finished" isn't being handled. The "stopped" state is handled:-

https://github.com/jellyfin/jellyfin-roku/blob/e6bfafb81d072d30ac874b268abc9333915b7d07/components/video/VideoPlayerView.bs#L511-L515

This state occurs when the back button is pressed, however when the video finishes naturally at the end the player does not generate a "stopped" state instead it generates a "finished" state which is never handled.

**Comment by roidy** [comment]
Can anybody else confirm this issue?

**Comment by cewert** [comment]
@roidy can you give #1941 a test when you get a chance?

**Comment by roidy** [comment]
I can confirm that build #1941 sends the correct finished state to the server.

However it does not fix this issue.

**Comment by cewert** [comment]
@roidy This is my last shot. I think it should work 🤞 https://github.com/jellyfin/jellyfin-roku/actions/runs/11092851858?pr=1941

**Comment by roidy** [comment]
@cewert Thanks, but unfortunately the video player still does not exit back into SDR mode.

Now all that happens is when the video ends the app pops the scene stack one screen further back than it should ie playing from the preplay screen pops you back to the library, and worst of all playing from the home screen pops you back out of app altogether to the Roku home screen.

**Comment by cewert** [comment]
Understood, thank you for testing! I'll revert that last commit.

Can you provide a sample of one of the files you're seeing this with? I'll test with an HDR TV when I get a chance

**Comment by roidy** [comment]
This a a small test file I've been using.

Download page:- https://www.demolandia.net/downloads.html?id=543675873
Direct link:- https://www.demolandia.net/z2e2

But it happens on every HDR10, HDR10+ or Dolby Vision file I have. It doesn't happen when using the Roku's native media playing app:- https://channelstore.roku.com/en-gb/details/1829ae20b082957f963d7dab42abf7e5/roku-media-player

**Comment by roidy** [comment]
@cewert did you try to reproduce this issue?

**Comment by cewert** [comment]
No, I have a 1080p TV. Not sure when I'll have access to a HDR TV to test

**Comment by roidy** [comment]
OK, when you asked for a sample I assumed you had a HDR capable TV.

No worries, it's one of those issues that 99.9% of people wont ever run into, as most people will stop the video once they get to the credits.
